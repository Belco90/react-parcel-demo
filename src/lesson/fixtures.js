const bulbasaur = {
  name: 'bulbasaur',
  types: [{ type: { name: 'grass' } }, { type: { name: 'poison' } }],
  sprites: {
    front_default: 'https://pokemon.png',
  },
}

export { bulbasaur }
