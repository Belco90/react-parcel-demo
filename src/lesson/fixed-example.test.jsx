import React from 'react'
import { act, render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import CountDemo from '../demos/CountDemo'
import PokemonDemo from '../demos/PokemonDemo'
import WeatherDemo from '../demos/WeatherDemo'
import { bulbasaur, madridWeather } from './fixtures'

describe('Count demo', () => {
  beforeAll(() => {
    jest.useFakeTimers()
  })

  afterAll(() => {
    jest.useRealTimers()
  })

  it('should count the number of clicks', () => {
    render(<CountDemo />)

    const countButton = screen.getByRole('button')
    expect(countButton).toHaveTextContent('Count is: 0')

    userEvent.click(countButton)

    expect(countButton).toHaveTextContent('Count is: 1')
  })

  it('should count how long the page was open', () => {
    render(<CountDemo />)

    expect(
      screen.getByText('Page has been open for 0 seconds.'),
    ).toBeInTheDocument()

    act(() => jest.advanceTimersByTime(1000))

    expect(
      screen.getByText('Page has been open for 1 seconds.'),
    ).toBeInTheDocument()
  })
})

describe('Pokémon demo', () => {
  const fetchMock = jest.fn().mockResolvedValue({
    ok: true,
    json: jest.fn().mockResolvedValue(bulbasaur),
  })

  beforeEach(() => {
    global.fetch = fetchMock
  })

  it('should display the basic Pokémon info', async () => {
    render(<PokemonDemo />)

    userEvent.type(screen.getByLabelText('Pick one Pokémon by their ID'), '1')
    userEvent.click(screen.getByRole('button'))

    expect(await screen.findByText('bulbasaur')).toBeInTheDocument()
  })
})

describe('Weather demo', () => {
  const fetchMock = jest.fn().mockResolvedValue({
    ok: true,
    json: jest.fn().mockResolvedValue(madridWeather),
  })

  beforeEach(() => {
    global.fetch = fetchMock
  })

  it('should display Madrid weather', async () => {
    render(<WeatherDemo />)

    await waitFor(() => expect(fetchMock).toHaveBeenCalledTimes(1))
    expect(screen.getByText('Temperature: 28.1ºC')).toBeInTheDocument()
  })
})
