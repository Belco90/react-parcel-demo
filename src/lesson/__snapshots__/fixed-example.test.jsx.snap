// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the basic Pokémon info 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <form>
      <label>
        Pick one Pokémon by their ID
        <div>
          <input
            max="898"
            min="1"
            name="pokemonId"
            required=""
            type="number"
          />
        </div>
        <button
          type="submit"
        >
          Search
        </button>
      </label>
    </form>
    <div>
      <img
        alt="Sprite of bulbasaur"
        src="https://pokemon.png"
      />
      <ul>
        <li>
          <strong>
            Name:
          </strong>
           
          bulbasaur
        </li>
        <li>
          <strong>
            Types:
          </strong>
           
          grass, poison
        </li>
      </ul>
    </div>
  </header>
</div>
`;
